apps/sasquatch/src/main/java/com/microsoft/appcenter/sasquatch/activities/SettingsActivity.java

```diff
@@ -494,23 +494,46 @@ public boolean onPreferenceChange(Preference preference, Object newValue) {
                     return true;
                 }
             });
-            initEditText(R.string.app_secret_key, R.string.app_secret_title, APP_SECRET_KEY, getString(R.string.app_secret), new EditTextListener() {
+            final String appSecret = MainActivity.sSharedPreferences.getString(APP_SECRET_KEY, "9e0d97c1-7838-46d0-9dab-1a0ef66aec6e");
+            initChangeableSetting(R.string.app_secret_key, appSecret, new Preference.OnPreferenceChangeListener() {
 
                 @Override
-                public void onSave(String value) {
+                public boolean onPreferenceChange(Preference preference, Object newValue) {
+                    if (newValue == null) {
+                        return true;
+                    }
+                    String value = newValue.toString();
+
                     if (!TextUtils.isEmpty(value)) {
-                        setKeyValue(APP_SECRET_KEY, value);
-                        Toast.makeText(getActivity(), String.format(getActivity().getString(R.string.app_secret_changed_format), value), Toast.LENGTH_SHORT).show();
+                        if (value.equals("Custom")) {
+                            /* If user selected custom, popup a text editor so they can enter whatever they want */
+                            ShowPreferenceTextEditor(getPreferenceManager().findPreference(getString(R.string.app_secret_key)), R.string.app_secret_title, APP_SECRET_KEY, getString(R.string.app_secret), new EditTextListener() {
+                                @Override
```

Comment:
```suggestion

                                @Override
```

Always a new line before annotations.